### PR TITLE
No blocks adjacent to cloud issue#42

### PR DESCRIPTION
**Pull request by bakursait2**
Solve [Issue#42](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/42)
* Add a new feature: Users cannot add blocks to the clouds.
  * When a user right-clicks the mouse while pointing at clouds, no action is taken.
* Functions used:
  * `on_mouse_press()`: A built-in function in pyglet that triggers an action when the user clicks the mouse.
  * When the user clicks the mouse, the function determines whether to call `add_block()` or not. If the block is of type cloud, `add_block()` is not called. Otherwise, `add_block()` is called.
  
  **Tests:**
* Include tests to verify that blocks cannot be added to clouds.

**Embedded Design Documentation:**
* Enhance the documentation of functions related to the clouds feature to ensure compatibility with `Doxygen`.
* Add the `issue#` number to each function related to the clouds feature, including `test_cloud.py`.

### CODE REVIEW

File: tempus_fugit_minecraft/model.py [hunk 1]
```diff
@@ -134,15 +134,12 @@ def exposed(self, position: tuple) -> bool:
         return False
 
     def add_block(self, position: tuple, texture: list, immediate=True) -> None:
-        """Add a block with the given `texture` and `position` to the world.
+        """!
```
Comment (w1158mxs) [inline tempus_fugit_minecraft/model.py:137]: What does this mean? (!)

File: tempus_fugit_minecraft/model.py [hunk 2]
```diff
@@ -348,21 +345,16 @@ def process_entire_queue(self) -> None:
         while self.queue:
             self._dequeue()
 
+    #issue20; #issue28
```
Comment (w1158mxs) [inline tempus_fugit_minecraft/model.py:348]: Good one!

**Review by w1158mxs — APPROVED** [review]
What about light and dark clouds, If we add them for users to build them is this will make a conflict?

File: tempus_fugit_minecraft/model.py (hunk 1 above)
Comment (bakursait2) [reply tempus_fugit_minecraft/model.py:137]: The exclamation mark or `!` is what to tell `DOXYGEN` how to treat the comments and added them to the documentation.

**Review by xieWright — APPROVED** [review]
Seems to be mostly adding in the Doxygen functionality. Very good.

**Review by Lurk390 — APPROVED** [review]
The logic and doxygen changes look good to me!

**Review by criticalglitch — APPROVED** [review]
Fundamentally, this directly conflicts with #68. The biggest indication for that is that we're pulling in both a window and a model into the cloud tests. Clouds and block placement decisions are pure game logic decisions which ultimately **belong in the model**, so we shouldn't need a Window to make those kinds of decisions.

However since that issue is still open, and I haven't pushed a PR to fix it yet. I don't see a reason to not approve this as is.

**Comment by bakursait2** [comment]
I took a screenshot of the test code result. 
I ran the command: `poetry run pytest --cov` and it shows the following:

image

**Pull request merged**